packages/back-end/src/util/features.ts

```diff
@@ -0,0 +1,51 @@
+import { SavedGroupInterface } from "../../types/saved-group";
+import { GroupMap, AttributeMap } from "../services/features";
+
+function getAttributeType(
+  allGroups: SavedGroupInterface[] | [],
+  groupId: string,
+  attributeMap: AttributeMap
+) {
+  const index = allGroups?.findIndex(
+    (group: SavedGroupInterface) => group.id === groupId
+  );
+  const attribute = allGroups[index].attributeKey;
+  const type: string | undefined = attributeMap?.get(attribute);
+
+  return type;
+}
+
+export function replaceSavedGroupsInCondition(
+  condition: string,
+  groupMap: GroupMap,
+  allGroups: SavedGroupInterface[] | [],
+  attributeMap: AttributeMap
+) {
+  const newString = condition
+    .replace(
+      // Ex: replace { $inGroup: "sdf8sd9f87s0dfs09d8" } with { $in: ["123, 345, 678, 910"]}
+      /"\$inGroup"[\s|\n]*:[\s|\n]*"([^"]*)"/g,
+      (match: string, groupId: string) => {
+        const type = getAttributeType(allGroups, groupId, attributeMap);
```

Comment:
@jdorn I made some changes here - they feel a little messy right now, but I found a bug where, if a user builds a savedGroup based on a `number` type attribute, we were always serving the user an array of numbers for the feature rule, instead of an array of numbers.

I'd love some insight on if you think there is a cleaner way to do this.

Essentially, I built an `attributeMap` that maps a property to a type, and then during the transform, I'm getting the `attributeKey` of each `groupId` and getting the type for that `attributeKey` and if it's a number, I'm calling parseFloat on each index.

For a normal `$in` rule for number attributes, we call parseFloat on the array BEFORE we store it in the feature's rules array, so this is a little different.

Reply:
I added a comment above that addresses this